### ISSUE TITLE
Getting an init issue with unexpected keyword argument 'method_whitelist'

### ISSUE THREAD

**Issue by poggdogg** [georgestarcher/Splunk-Class-httpevent#29]
Only in windows, not on all windows machines either.
Running on the local host:
```
Retry.__init__() got an unexpected keyword argument 'method_whitelist'
Traceback (most recent call last):
  File "E:\Program Files\Python\Python311\Lib\site-packages\splunk_http_event_collector.py", line 250, in _batchThread
    response = self.requests_retry_session().post(self.server_uri, data=payload, headers=headers, verify=self.SSL_verify)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Program Files\Python\Python311\Lib\site-packages\splunk_http_event_collector.py", line 69, in requests_retry_session
    retry = Retry(total=retries, read=retries, connect=retries, backoff_factor=backoff_factor, status_forcelist=status_forcelist, method_whitelist=frozenset(['HEAD', 'TRACE', 'GET', 'PUT', 'OPTIONS', 'DELETE', 'POST']))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Retry.__init__() got an unexpected keyword argument 'method_whitelist'
Retry.__init__() got an unexpected keyword argument 'method_whitelist'
Traceback (most recent call last):
  File "E:\Program Files\Python\Python311\Lib\site-packages\splunk_http_event_collector.py", line 250, in _batchThread
    response = self.requests_retry_session().post(self.server_uri, data=payload, headers=headers, verify=self.SSL_verify)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Program Files\Python\Python311\Lib\site-packages\splunk_http_event_collector.py", line 69, in requests_retry_session
    retry = Retry(total=retries, read=retries, connect=retries, backoff_factor=backoff_factor, status_forcelist=status_forcelist, method_whitelist=frozenset(['HEAD', 'TRACE', 'GET', 'PUT', 'OPTIONS', 'DELETE', 'POST']))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Retry.__init__() got an unexpected keyword argument 'method_whitelist'
```

**Comment by xpac1985** [comment]
Some background: https://github.com/urllib3/urllib3/issues/2092

tl;dr: `method_whitelist` is now `allowed_methods`.
